### PR TITLE
Add jaccard loss

### PR DESCRIPTION
**Pull request by devisperessutti**
PR that adds Jaccard distance (Intersection over Union) loss.

This distance can be useful in semantic segmentation tasks.

### CODE REVIEW

File: eoflow/tasks/train.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class TrainAndEvaluateTask(Schema):
         train_input_config = fields.Nested(nested=ObjectConfiguration, required=True, description="Input type and configuration for training.")
         val_input_config = fields.Nested(nested=ObjectConfiguration, required=True, description="Input type and configuration for validation.")
 
-        validation_steps = fields.Int(missing=None, description="Number of batches to compute validation scores over")
+        validation_steps = fields.Int(missing=1, description="Number of batches to compute validation scores over")
```
Comment (lojzezust) [inline eoflow/tasks/train.py]: Same as above.

File: eoflow/models/losses.py [hunk 2]
```diff
@@ -50,3 +50,45 @@ def call(self, y_true, y_pred):
...
+        if self.from_logits:
+            y_pred = tf.nn.softmax(y_pred)
+
+        intersection = tf.reduce_sum(y_true * y_pred, axis=(1, 2))
+
+        sum_ = tf.reduce_sum(y_true + y_pred, axis=(1, 2))
+
+        jac = (intersection + self.smooth) / (sum_ - intersection + self.smooth)
+
+        loss = tf.reduce_mean((1 - jac) * self.smooth)
```
[the remark this answers is not in the document]
Comment (lojzezust) [reply eoflow/models/losses.py]: I see this was addressed already in #13.

File: eoflow/models/segmentation.py [hunk 3]
```diff
@@ -240,6 +241,7 @@ def train_and_evaluate(self,
                            num_epochs,
                            iterations_per_epoch,
                            model_directory,
+                           validation_steps=1,
```
[the remark this answers is not in the document]
Comment (devisperessutti) [reply eoflow/models/segmentation.py]: which defaults should we use? 

by using `None` I was getting an error that step is required for infinite tf datasets

[the remark this answers is not in the document]
Comment (lojzezust) [reply eoflow/models/segmentation.py]: Oh, my bad. Didn't realize at first you added the `validation_steps` parameter. I was thinking about `iterations_per_epoch`.

In your case the problem seems to be, that the `val_dataset` you provided is infinite in which case the evaluation can't be performed. For the validation always a finite dataset must be given. It's better to resolve this on the dataset end not here (by ceating a finite dataset or calling `dataset.take(number_of_examples)`). You can call `train_and_evaluate` with the `validation_steps` parameter if other options are not available, but it should not have a default value.

[the remark this answers is not in the document]
Comment (devisperessutti) [reply eoflow/models/segmentation.py]: so you suggest removing the `validation_steps` altogether?

[the remark this answers is not in the document]
Comment (lojzezust) [reply eoflow/models/segmentation.py]: Yeah I think that would be the best approach. It can still be specified manualy if needed, as it will be passed through `**kwargs` to the `fit` method.

[the remark this answers is not in the document]
Comment (devisperessutti) [reply eoflow/models/segmentation.py]: OK, `validation_steps` removed, and made `validation_set` not repeating

File: eoflow/tasks/train.py (hunk 1 above)
Comment (devisperessutti) [reply eoflow/tasks/train.py]: resolved as above
